src/main/java/com/commercetools/payment/notification/CommercetoolsPaymentNotificationController.java

```diff
@@ -40,17 +43,22 @@
 
     private final Logger logger;
 
+    private final Gson paypalGson;
+
     @Autowired
     public CommercetoolsPaymentNotificationController(@Nonnull StringTrimmerEditor stringTrimmerEditor,
-                                                      @Nonnull NotificationEventDispatcherProvider eventDispatcherProvider) {
+                                                      @Nonnull NotificationEventDispatcherProvider eventDispatcherProvider,
+                                                      @Nonnull Gson paypalGson) {
         super(stringTrimmerEditor);
         this.eventDispatcherProvider = eventDispatcherProvider;
+        this.paypalGson = paypalGson;
         logger = LoggerFactory.getLogger(PaymentHandler.class);
     }
 
     @PostJsonRequestJsonResponseMapping(value = "/{tenantName}/" + NOTIFICATION_PATH_URL)
     public CompletionStage<ResponseEntity> handleNotification(@PathVariable String tenantName,
-                                                              @RequestBody PaypalPlusNotificationEvent eventFromPaypal) {
+                                                              @Nonnull HttpServletRequest request) throws IOException {
+        PaypalPlusNotificationEvent eventFromPaypal = paypalGson.fromJson(getBody(request), PaypalPlusNotificationEvent.class);
```

Comment:
Pls, try to avoid `PaypalPlusNotificationEvent` with those weird names. Likely `paypalGson` shell correctly map underscores to camel case

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/78/commits/a69fdb26fe8bc6fa58b96db294f6118ba320248a